### PR TITLE
Migrate exportdb to torch.export from torchdynamo.export

### PR DESCRIPTION
**Pull request by tugsbayasgalan**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103861
* #102310

Things that needed to be fixed:
1. Fix a bug with returning dict output type 
2. Make pass_base work with map implementation
3. Fix subtle bug with dynamo not propagating "val" in node.meta
4. Add export_constraints field in ExportCase in ExportDB

cc @voznesenskym @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @ipiszy @chenyang78

### CODE REVIEW

File: torch/_export/pass_base.py [hunk 1]
```diff
@@ -326,16 +326,16 @@ def call_cond(
...
     ) -> ProxyValue:
-        f_branch = self.call_submodule(f, tuple([xs.data[0]] + list(args)))
+        f_branch = self.call_submodule(f, (args[:num_args][0], *args[num_args:]))
         assert f_branch is not None
         return self._fx(
             "call_function",
-            control_flow.map,
-            (f_branch.graph_module, xs, *args),
+            torch.ops.map_impl,
+            (f_branch.graph_module, num_args, *args),
```
Comment (tugsbayasgalan) [inline torch/_export/pass_base.py:338]: @ydwu4 Is this correct?

File: torch/_export/db/case.py [hunk 2]
```diff
@@ -77,6 +78,7 @@ class ExportCase:
     # Tags associated with the use case. (e.g dynamic-shape, escape-hatch)
     tags: Set[str] = field(default_factory=lambda: set())
     support_level: SupportLevel = SupportLevel.SUPPORTED
+    export_constraints: List[Constraint] = field(default_factory=lambda: [])
```
Comment (zhxchen17) [inline torch/_export/db/case.py:81]:
```suggestion
    constraints: List[Constraint] = field(default_factory=lambda: [])
```

**Review by zhxchen17 — APPROVED** [review]
thanks!

**Review by ydwu4 — APPROVED** [review]

File: torch/_export/pass_base.py (hunk 1 above)
Comment (tugsbayasgalan) [reply torch/_export/pass_base.py:338]: ok sounds good

**Pull request closed without merging**